core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsFunction.scala

```diff
@@ -340,17 +354,19 @@ object InterpretExprAsFunction {
 
   final case class Input[F[_], V](
     value: F[V],
-    evidence: Evidence,
+    evidence: Evidence, // TODO: Is this needed? Can't I just add the input value as evidence when it is made of Facts?
```

Comment:
This seems needed because it must be possible to select a value (like `0.8`) from a fact (like `ProbabilityToUse().scores("weightloss")`) without losing the fact it came from.

My hope was that I would be able to combine the evidence at a higher level in the expression once the value has been used, but I feel like there are too many edge cases to handle there and I'm not sure it is possible. However, with the current input.evidence, it is hard to know when to use it. I think there just needs to be a lot more unit tests around expected behavior before I'll know whether my hypothesis that we can handle this at some higher level expression can be proven or disproven.

Reply:
I'm gonna remove this comment for now as I am doubting its possibility again and it isn't relevant to this PR.